rest_framework_dataclasses/types.py

```diff
@@ -0,0 +1,32 @@
+"""Type aliases/definitions for type checkers"""
+from typing import TYPE_CHECKING, Type, Union, List, Tuple, Any, Dict
+
+import rest_framework.fields
+
+
+# Define some types to make type hinting more readable
+KWArgs = Dict[str, Any]
+SerializerField = rest_framework.fields.Field
+SerializerFieldDefinition = Tuple[Type[SerializerField], KWArgs]
+FieldsType = Union[List[str], Tuple[str]]
+
+if TYPE_CHECKING:
+    from typing import Protocol, Literal
+
+    FieldsOrAllType = Union[FieldsType, Literal['__all__']]
+
+    class _MetaProtocol(Protocol):
+        """
+        Type annotation for Serializer `Meta` attribute. `Protocol` indicates that it's a duck type, not parent class.
+        """
+        dataclass: type
+        fields: FieldsOrAllType
+        exclude: FieldsType
+        read_only_fields: FieldsType
+
+    Metaclass = Type[_MetaProtocol]
+
+else:
```

Comment:
This would need a real check for Python 3.7, since TYPE_CHECKING can also be true there.

Reply:
Technically not necessary, because types aren't needed at runtime, and at type-check-time, mypy has internal knowledge of `typing.Literal` even on Python 3.7 or older.

But I don't mind either way, will change.

Reply:
> mypy has internal knowledge of typing.Literal even on Python 3.7 or older.

Ah, I didn't know that.